src/components/visualizations/common/utils/drilldownEventing.ts

```diff
@@ -0,0 +1,205 @@
+// (C) 2007-2018 GoodData Corporation
+import { get, debounce } from 'lodash';
+import * as invariant from 'invariant';
+import * as CustomEvent from 'custom-event';
+import { IDrillableItem } from '../../../../';
+import { AFM, Execution } from '@gooddata/typings';
+import { IDrillEventIntersectionElement } from '../../../../interfaces/DrillEvents';
+import { VisElementType, VisType, VisualizationTypes } from '../../../../constants/visualizationTypes';
+
+export interface IDrillableItemLocalId extends IDrillableItem {
+    localIdentifier: AFM.Identifier;
+}
+
+export function isDrillableItemLocalId(item: IDrillableItem | IDrillableItemLocalId): item is IDrillableItemLocalId {
+    return (item as IDrillableItemLocalId).localIdentifier !== undefined;
+}
+
+export interface IDrillIntersection {
+    id: string;
+    title: string;
+    value: Execution.DataValue;
+    name: string;
+    uri: string;
+    identifier: AFM.Identifier;
+}
+
+export interface IDrillPoint {
+    x: number;
+    y: number;
+    drillContext: IDrillIntersection[];
+}
+
+export interface IHCDrillEventGroup {
+    points: IDrillPoint[];
+}
+export interface IHCDrillEventPoint {
+    point: IDrillPoint;
+}
+
+export interface ICellDrillEvent {
+    columnIndex: number;
+    rowIndex: number;
+    row: string[];
+    intersection: IDrillIntersection[];
+}
+
+export type FiredDrillEventCallback = (data: any) => boolean;
+
+export interface IDrillConfig {
+    afm: AFM.IAfm;
+    onFiredDrillEvent: FiredDrillEventCallback;
+}
+
+export function isGroupHCDrillEvent(event: IHCDrillEventGroup | IHCDrillEventPoint): event is IHCDrillEventGroup {
```

Comment:
what is `IHCDrillEventGroup` ? or `IHCDrillEventPoint` ?

Reply:
Those are event that produces Highcharts. I will include Highcharts typings and use its interfaces instead.